In [1]:
#importing necessary libraries
import pandas as pd
import numpy as np
import seaborn as sns
import scipy.stats as stat

In [2]:
#reading the csv file
data=pd.read_csv('Sales_add.csv')
data

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [3]:
#describtion about the data
data.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

In [4]:
#looking if there is any null value
data.isna().sum()

Month                             0
Region                            0
Manager                           0
Sales_before_digital_add(in $)    0
Sales_After_digital_add(in $)     0
dtype: int64

# The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

Here we are using t-test since the sample size<30.

HO=Null Hypothesis

H1=Alternative Hypothesis


### H0:There is no increase in sales after stepping into digital marketing.

### H1:There is increase in sales after stepping into digital marketing.  

In statistical tests, statistical significance is determined by citing an alpha level, or the probability of rejecting the null hypothesis when the null hypothesis is true. In most cases, researchers use an alpha of 0.05, which means that there is a less than 5% chance that the data being tested could have occurred under the null hypothesis.

In [5]:
#input alpha_value
alpha_value=float(input("alpha_value:"))

alpha_value:0.05


In [6]:
#using t test to find the p value
#since we need to compare sales before and after stepping to digital marketing,we are using data['Sales_before_digital_add(in $)'],data['Sales_After_digital_add(in $)'] as parameters 

ttest,p_value=stat.ttest_rel(data['Sales_before_digital_add(in $)'],data['Sales_After_digital_add(in $)'])
print(p_value)

6.336667004575778e-11


In [7]:
#comparing the values and printing the result
if p_value<alpha_value:
    print("we reject the null hypothesis since p_value<alpha value i.e ",p_value,"<",alpha_value)
else:
    print("we accept the null hypothesis since p_value>alpha value i.e ",p_value,">",alpha_value)

we reject the null hypothesis since p_value<alpha value i.e  6.336667004575778e-11 < 0.05


Here we rejected the null hypothesis.So we can conclude that there is increase in sales after stepping into digital marketing 

# The company needs to check whether there is any dependency between the features “Region” and “Manager”.

Here we have to check the dependency between 2 features.So we use the chi square test  for Independence.

HO=Null Hypothesis

H1=Alternative Hypothesis


### H0:Region and Manager are independent to each other

### H1:Region and Manager are dependent to each other

In [8]:
alpha_value=float(input("alpha_value:"))

alpha_value:0.05


In [9]:
#passing a crosstabulation table
crosstab = pd.crosstab(data["Region"], data["Manager"])
crosstab

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [11]:
chi_square,p_value,DOF,array=stat.chi2_contingency(crosstab)

 The information is returned within a tuple where the first value is the chi-square test static, the second value is the p-value, and the third number is the degrees of freedom and an array is also returned which contains the expected cell counts.

In [12]:
#display chi_square value
chi_square

3.050566893424036

In [13]:
#display degrees of freedom
DOF

4

In [14]:
#dispaly p_value
p_value

0.5493991051158094

In [15]:
#comparing the values and printing the result
if p_value<alpha_value:
    print('We reject the null hypothesis since p_value<alpha value i.e ',p_value,"<",alpha_value)
else:
    print('We accept the null hypothesis since p_value>alpha value i.e ',p_value,">",alpha_value)

We accept the null hypothesis since p_value>alpha value i.e  0.5493991051158094 > 0.05


Also the calculated chi square value 3.0506,which is less than the chi square table value 9.488(for alpha value=0.05 with degrees of freedom=4 ),so we accept the null hypothesis.

So we can conclude that the features Region and Manager are not dependent

### We can also conduct the chi-square test of independence using Researchpy

In [16]:
#installing researchpy to import it

In [17]:
pip install researchpy

Note: you may need to restart the kernel to use updated packages.


In [18]:
#importing researchpy
import researchpy as rp

In [19]:
#Returns a data table that includes the counts and percentages of each category. 
rp.summary_cat(data[["Region", "Manager"]])

Variable      Outcome  Count  Percent
0   Region   Region - A     10    45.45
1            Region - B      7    31.82
2            Region - C      5    22.73
3  Manager  Manager - A      9    40.91
4           Manager - B      7    31.82
5           Manager - C      6    27.27

In [20]:
crosstab,test_results,expected=rp.crosstab(data.Region,data.Manager,test="chi-square",expected_freqs= True,prop="cell")
test_results

Chi-square test  results
0  Pearson Chi-square ( 4.0) =    3.0506
1                    p-value =    0.5494
2                 Cramer's V =    0.2633

Since p_value>alpha_value,we fail to reject the null hypothesis.So features Region and Manager are independent

In [21]:
expected

Manager                        
Manager    Manager - A Manager - B Manager - C
Region                                        
Region - A    4.090909    3.181818    2.727273
Region - B    2.863636    2.227273    1.909091
Region - C    2.045455    1.590909    1.363636